api/v1beta1/azuremanagedcluster_webhook_test.go

```diff
@@ -135,27 +136,29 @@ func TestAzureManagedCluster_ValidateCreate(t *testing.T) {
 
 func TestAzureManagedCluster_ValidateCreateFailure(t *testing.T) {
 	tests := []struct {
-		name        string
-		amc         *AzureManagedCluster
-		deferFunc   func()
-		expectError bool
+		name               string
+		amc                *AzureManagedCluster
+		featureGateEnabled *bool
+		expectError        bool
 	}{
 		{
-			name:        "feature gate explicitly disabled",
-			amc:         getKnownValidAzureManagedCluster(),
-			deferFunc:   utilfeature.SetFeatureGateDuringTest(t, feature.Gates, capifeature.MachinePool, false),
-			expectError: true,
+			name:               "feature gate explicitly disabled",
+			amc:                getKnownValidAzureManagedCluster(),
+			featureGateEnabled: ptr.To(false),
+			expectError:        true,
 		},
 		{
-			name:        "feature gate implicitly enabled",
-			amc:         getKnownValidAzureManagedCluster(),
-			deferFunc:   func() {},
-			expectError: false,
+			name:               "feature gate implicitly enabled",
+			amc:                getKnownValidAzureManagedCluster(),
+			featureGateEnabled: nil,
+			expectError:        false,
 		},
 	}
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			defer tc.deferFunc()
+			if tc.featureGateEnabled != nil {
+				defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, capifeature.MachinePool, *tc.featureGateEnabled)()
```

Comment:
The main change here is ensuring the `*testing.T` handle passed to `SetFeatureGateDuringTest` is the one for the individual subtest. I wonder if the change to enable the feature gate by default in CAPI is what caused this to start failing.